oap-server/server-library/library-util/src/main/java/org/apache/skywalking/oap/server/library/util/MultipleFilesChangeMonitor.java

```diff
@@ -0,0 +1,242 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.library.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * MultipleFilesChangeMonitor provides the capability to detect file or multiple files changed. It provide second level
+ * change detection and feedback mechanism.
+ *
+ * Due to memory cost, this monitor mechanism is not suitable for small files and usually being changed on the runtime
+ * by user manually or 3rd party tool. Typical, these files are config information or authentication files.
+ */
+@Slf4j
+public class MultipleFilesChangeMonitor {
+    /**
+     * The backend scheduler to trigger all file monitoring.
+     */
+    private static ScheduledFuture<?> FILE_MONITOR_TASK_SCHEDULER;
+    private static ReentrantLock SCHEDULER_CHANGE_LOCK = new ReentrantLock();
+    /**
+     * The list contains all monitors.
+     */
+    private static List<MultipleFilesChangeMonitor> MONITOR_INSTANCES = new ArrayList<>();
+
+    /**
+     * The timestamp when last time do status checked.
+     */
+    private long lastCheckTimestamp = 0;
+    /**
+     * The period of watching thread checking the file status. Unit is the second.
+     */
+    private final long watchingPeriodInSec;
+    private List<WatchedFile> watchedFiles;
+    private FilesChangedNotifier notifier;
+
+    /**
+     * Create a new monitor for the given files
+     *
+     * @param watchingPeriodInSec The check period.
+     * @param notifier            to accept the file changed notification.
+     * @param files               to be monitored.
+     */
+    public MultipleFilesChangeMonitor(long watchingPeriodInSec,
+                                      FilesChangedNotifier notifier,
+                                      String... files) {
+        watchedFiles = new ArrayList<>();
+        this.watchingPeriodInSec = watchingPeriodInSec;
+        this.notifier = notifier;
+        for (final String file : files) {
+            WatchedFile monitor = new WatchedFile(file);
+            watchedFiles.add(monitor);
+        }
+    }
+
+    /**
+     * Check file changed status, if so, send the notification.
+     */
+    private void checkAndNotify() {
+        if (System.currentTimeMillis() - lastCheckTimestamp < watchingPeriodInSec * 1000) {
+            // Don't reach the period threshold, ignore this check.
+            return;
+        }
```

Comment:
Why do we need to double-check? To avoid the new added Monitor will re-check at once?
IMO, we can set the check period as `watchingPeriodInSec`, because this case is a minority case and only affects the first check.

Reply:
Because the case is, the timer period is certain and shorter than watcher period. We need to make sure the check period follows the user requirements.

Reply:
I got it. Monitors own their private schedule, `watchingPeriodInSec`.